src/lib/scene-internal.ts

```diff
@@ -455,38 +460,41 @@ export class SceneInternal implements Renderer {
       throw new Error('Could not find REGL');
     }
 
+    this.canvas = document.createElement('canvas');
+    Object.assign(this.canvas.style, {
+      border: 0,
+      height: '100%',
+      left: 0,
+      margin: 0,
+      padding: 0,
+      top: 0,
+      width: '100%',
+    });
+    this.container.appendChild(this.canvas);
+
+    const bounds = this.canvas.getBoundingClientRect();
+    const width = bounds.right - bounds.left;
+    const height = bounds.bottom - bounds.top;
```

Comment:
Why not just use the `DOMRect`'s [`width` and `height` properties](https://developer.mozilla.org/en-US/docs/Web/API/DOMRect#properties)?

```suggestion
    const {width, height} = this.canvas.getBoundingClientRect();
```

Reply:
Good question! I noticed this method of doing things in the regl codebase ([webgl.js line 30](https://github.com/regl-project/regl/blob/master/lib/webgl.js#L31)):

```js
var bounds = canvas.getBoundingClientRect()
w = bounds.right - bounds.left
h = bounds.bottom - bounds.top
```

Doing some digging just now, I found that according to the W3C draft spec, the `width` and `height` could be negative. I believe that taking the difference of bounds will always yield a non-negative value. https://drafts.fxtf.org/geometry/#DOMRect

Having said that, I don't know under what circumstances the `width` or `height` would actually be negative. It may never occur. Probably best to just use actual `width`/`height` until we encounter a problem in the wild. I'll change it.

Reply:
I found the circumstances under which `width` and `height` might be less than 0: Flexbox! 

We found this [SO question](https://stackoverflow.com/questions/26465745/ellipsis-in-flexbox-container) when trying to figure out how to get text overflow working in LIT's color legend. 

In short, `display: flex;` can cause the browser to compute a negative `width` or `height` (though I believe only one at a time) for an element because `min-width` and `min-height` are unset by default. The fix is to style your `flex`-child elements to have a `min-*` property, though nested `flex` containers induce a pseudo-recursive problem where every `flex`-child in the trees needs to have a `min-*` property, so a [universal selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Universal_selectors) that sets minimum sizes to 0 might be best.

Reply:
Wow, thanks! If this turns out to be a problem in practice with Megaplot, please open an issue so we can track it. Until then, I'm happy to stick with this.